leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteAttributesRequest.java

```diff
@@ -16,58 +16,67 @@
 package org.eclipse.leshan.core.request;
 
 import org.eclipse.leshan.ObserveSpec;
+import org.eclipse.leshan.core.attributes.AttributeSet;
 import org.eclipse.leshan.core.node.LwM2mPath;
 import org.eclipse.leshan.core.request.exception.InvalidRequestException;
 import org.eclipse.leshan.core.response.WriteAttributesResponse;
 
 public class WriteAttributesRequest extends AbstractDownlinkRequest<WriteAttributesResponse> {
 
-    private final ObserveSpec observeSpec;
+    private final AttributeSet attributes;
 
-    public WriteAttributesRequest(int objectId, ObserveSpec observeSpec) throws InvalidRequestException {
-        this(new LwM2mPath(objectId), observeSpec);
+    public WriteAttributesRequest(int objectId, AttributeSet attributes) {
```

Comment:
About the InvalidRequestException... I thought that the convention was to only add throws if it is necessary (checked exceptions?), but I may very well be wrong about this.

Reply:
I think most people are doing like you thought. I mean "adding throws only for checked exceptions". I also think that most people use checked exception for "normal" flow and so force their users to catch exception. (which could make sense)

In Leshan, there are some cases where we choose to use "unchecked/runtime" Exception for exception which could make sense to catch in a normal execution flow. We don't want to force users to catch this exception but we would also give it visibility as we thought that this could be useful to catch it.

I don't know if it's clear :) but this is not a "so strange" way as even in the JDK code you could find something like this (e.g. Integer.parseInt)

Reply:
I will reinstate them. Our linter tells me to remove them, though... :)